compiler/luci/export/src/CircleTensorExporter.cpp

```diff
@@ -247,12 +247,21 @@ flatbuffers::Offset<Vector<int32_t>> encodeShape(FlatBufferBuilder &builder,
 }
 
 flatbuffers::Offset<Vector<int32_t>> encodeShapeSignature(FlatBufferBuilder &builder,
-                                                          const ShapeSignature &shape_signature)
+                                                          const ShapeDescription &shape)
 {
-  if (shape_signature.rank() == 0)
-    return 0;
+  assert(shape._rank_known && "unknown number of dimensions is not supported");
+  flatbuffers::Offset<Vector<int32_t>> signature_offset;
+  bool has_unknown = false;
+
+  for (uint32_t i = 0; i < shape._dims.size(); ++i)
+    if (shape._dims.at(i) == -1)
+      has_unknown = true;
 
-  return builder.CreateVector(shape_signature.as_vector());
+  // If all of dimensions are known, shape_signature should be empty.
+  if (has_unknown)
```

Comment:
above comment and the code seems to match in logic but explains in different view. can you match the code and the comment? current is a bit confusing.

Reply:
(optional)
```cxx
  // shape_signature is only set if one of dimensions is unknown.
  for (uint32_t i = 0; i < shape._dims.size(); ++i)
    if (shape._dims.at(i) == -1)
    {
      signature_offset = builder.CreateVector(shape._dims);
      break;
    }
```